### PR TITLE
Adds PDF and Excel comparator classes.

### PR DESCRIPTION
**Pull request by gmartinez3werfen**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            <name>Unlicense</name>
+            <url>http://unlicense.org</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <jasperrport.version>6.17.0</jasperrport.version>
```
Comment (joaquimvila) [inline pom.xml]: Why do we need to import jasperreport here? If I'm not wrong, to compare 2 PDF files, we just use pdfbox at this point. If so, I would suggest cleaning up this dependency.

File: src/main/java/com/werfen/report/test/excel/ExcelComparator.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+import static java.util.Objects.nonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ExcelComparator {
```
Comment (joaquimvila) [inline src/main/java/com/werfen/report/test/excel/ExcelComparator.java]: This is great, but we can't test that the code of this comparator is right unless we generate a release and use it in another project. Why not adding some minimal tests to check this comparison feature? They can rely on some given files and do things, like: 
* A file compared with himself passes the assert
* A file compared with another one fails the assert

Same applies for PDFs.

File: src/main/java/com/werfen/report/test/excel/ExcelComparator.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package com.werfen.report.test.excel;
```
Comment (joaquimvila) [inline src/main/java/com/werfen/report/test/excel/ExcelComparator.java]: No related to here, but I miss adding some continuous integration to guarantee in the PRs that this works as expected. I would suggest using the same approach as in the java-generator-library (using GitHub Actions)

**Review by joaquimvila — APPROVED** [review]
Good starting point for this lib. Added some comments to continue improving it.

File: pom.xml (hunk 1 above)
Comment (gmartinez3werfen) [reply pom.xml]: Good catch, I forgot to remove it.

File: src/main/java/com/werfen/report/test/excel/ExcelComparator.java (hunk 3 above)
Comment (gmartinez3werfen) [reply src/main/java/com/werfen/report/test/excel/ExcelComparator.java]: I'm not following you in that point. Can we discuss about it?

File: src/main/java/com/werfen/report/test/utils/excel/ExcelComparator.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            if (expectedCellCount != actualCellCount)
+                throw new FilesNotEqualException("Sheet " + actual.getSheet().getSheetName() + " have different number of cells in row " + actual.getRowNum(), String.valueOf(expectedCellCount), String.valueOf(actualCellCount));
+
+            for (int cellIndex = 0; cellIndex < expectedCellCount; cellIndex++) {
+                compareCellEquals(expected.getCell(cellIndex), actual.getCell(cellIndex));
+            }
+        }
+    }
+
+    public static void compareCellEquals(Cell expected, Cell actual) throws FilesNotEqualException {
```
Comment (alfonsserra) [inline src/main/java/com/werfen/report/test/utils/excel/ExcelComparator.java]: Could you please create a private method for each case?

**Comment by joaquimvila** [comment]
Thanks @alfonsserra and @gmartinez3werfen for the latest commits of this PR. For me, this is ready to be merged to main.

**Review by QuimNotoTrilla — APPROVED** [review]
Greate job

**Pull request merged**